### DISCUSSION TITLE
How to add tooltip on aggrid row?

### DISCUSSION THREAD

**Discussion by tz301** [zauberzeug/nicegui#2812] [Q&A] [answered]
### Question

From https://ag-grid.com/javascript-data-grid/tooltips/ , I can set tooltip using `headerTooltip` on column header.
But using `tooltipValueGetter` to set cell tooltip seems not working.

```py
ui.aggrid(
    {
        'columnDefs': [{'field': 'name', 'headerName': 'Name', 'headerTooltip': 'Tooltip for Column Header'}],
        'rowData': [{'name': 'Alice'}, {'name': 'Bob'}],
    }
)
```

![image](https://github.com/zauberzeug/nicegui/assets/35760938/5e227112-d703-4cce-a4d1-ef4b13aef5e8)

**Comment by falkoschindler** [comment]
Hi @tz301,

The "tooltipValueGetter" field needs to be a function. We can't instantiate JavaScript functions in Python, but we can send the function as a string. To indicate that the string needs to be converted back to a JavaScript instance, we use a colon ":" prefix:

```py
ui.aggrid({
    'columnDefs': [{
        'field': 'name',
        'headerName': 'Name',
        'headerTooltip': 'Tooltip for Column Header',
        ':tooltipValueGetter': '(params) => params.value.toUpperCase()',
    }],
    'rowData': [
        {'name': 'Alice'},
        {'name': 'Bob'},
    ],
})
```

**Reply by tz301** [reply]
@falkoschindler That works!

But if I want the tooltip to be selected from a dict, such as below the tooltip for 'Alice' is 'a', how can I write this?

```
name_mapping = {'Alice': 'a', 'Bob': 'b'}
':tooltipValueGetter': '(params) => name_mapping[params.value]',  # how to set this function?
```

**Reply by falkoschindler** [reply, marked as answer]
I'd recommend serializing and string-interpolating the `name_mapping` like this:
```py
name_mapping = {'Alice': 'a', 'Bob': 'b'}

ui.aggrid({
    'columnDefs': [{
        'field': 'name',
        'headerName': 'Name',
        'headerTooltip': 'Tooltip for Column Header',
        ':tooltipValueGetter': f'(params) => ({json.dumps(name_mapping)})[params.value]',
    }],
    'rowData': [
        {'name': 'Alice'},
        {'name': 'Bob'},
    ],
})
```

**Reply by tz301** [reply]
Thanks!
